### ISSUE TITLE
My llama.cpp cannot quantify Qwen1.5-MoE-A2.7B. Can any technical experts tell me how to solve it?

### ISSUE THREAD

**Issue by MRQJsfhf** [QwenLM/Qwen1.5#281]
Tried multiple instructions but still can't quantify

The situation is reported as follows:
llama.cpp # python convert.py models/MOE 
Loading model file models\MOE\model-00001-of-00008.safetensors
Loading model file models\MOE\model-00001-of-00008.safetensors
Loading model file models\MOE\model-00002-of-00008.safetensors
Loading model file models\MOE\model-00003-of-00008.safetensors
Loading model file models\MOE\model-00004-of-00008.safetensors
Loading model file models\MOE\model-00005-of-00008.safetensors
Loading model file models\MOE\model-00006-of-00008.safetensors
Loading model file models\MOE\model-00007-of-00008.safetensors
Loading model file models\MOE\model-00008-of-00008.safetensors
params = Params(n_vocab=151936, n_embd=2048, n_layer=24, n_ctx=8192, n_ff=5632, n_head=16, n_head_kv=16, n_experts=None, n_experts_used=None, f_norm_eps=1e-06, rope_scaling_type=None, f_rope_freq_base=1000000.0, f_rope_scale=None, n_orig_ctx=None, rope_finetuned=None, ftype=None, path_model=WindowsPath('models/MOE'))
Traceback (most recent call last):
  File "C:\Git\llama.cpp\convert.py", line 1548, in <module>
    main()
  File "C:\Git\llama.cpp\convert.py", line 1515, in main
    vocab, special_vocab = vocab_factory.load_vocab(vocab_types, model_parent_path)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Git\llama.cpp\convert.py", line 1417, in load_vocab
    vocab = self._create_vocab_by_path(vocab_types)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Git\llama.cpp\convert.py", line 1407, in _create_vocab_by_path
    raise FileNotFoundError(f"Could not find a tokenizer matching any of {vocab_types}")
FileNotFoundError: Could not find a tokenizer matching any of ['spm', 'hfft']

**Comment by jklj077** [comment]
The team is currently working on it. For the latest update, please refer to <https://github.com/ggerganov/llama.cpp/pull/6074>.

**Comment by jklj077** [comment]
Closed as the PR has been merged.

**Comment by ThiloteE** [comment]
I encountered the same error message, even though the PR has been merged. I tried with standard, then `--vocab-type bpe`, with `--pad-vocab` and finally `-skip-unknown`, all to no avail. Finally, I found some quants here: https://huggingface.co/qwp4w3hyb/Qwen1.5-MoE-A2.7B-Chat-iMat-GGUF/tree/main. How come?

**Comment by jklj077** [comment]
Hi, were you using convert-hf-to-gguf.py or convert.py? I think convert-hf-to-gguf.py is the one.

**Comment by ThiloteE** [comment]
I used convert.py, because in the past, I had successfully used that to convert pytorch-model.bin files to gguf. I now tried convert-hf-to-gguf.py for the Qwen1.5-MoE-A2.7B safetensors and it works. Here is the command I used to do the initial conversion: `python3.11 convert-hf-to-gguf.py models\Qwen1.5-MoE-A2.7B-Chat\ --outtype f16`. Thank you for the hint @jklj077. The quants in qwp4w3hyb's repository are functioning as well. 👍
